src/components/portal/portal.js

```diff
@@ -1,127 +1,95 @@
-      if (this.scrollParent) {
-        this.scrollParent.addEventListener("scroll", this.props.onReposition);
-      }
-      Browser.getWindow().addEventListener("resize", this.props.onReposition);
+  useEffect(() => {
+    if (onReposition) {
+      onReposition();
+      window.addEventListener("resize", onReposition);
     }
-  }
 
-  componentDidUpdate(prevProps) {
-    if (prevProps.onReposition !== this.props.onReposition) {
-      if (this.scrollParent) {
-        this.scrollParent.removeEventListener("scroll", prevProps.onReposition);
-        this.scrollParent.addEventListener("scroll", this.props.onReposition);
+    return () => {
+      if (onReposition) {
+        window.removeEventListener("resize", onReposition);
       }
-      Browser.getWindow().removeEventListener("resize", prevProps.onReposition);
-      Browser.getWindow().addEventListener("resize", this.props.onReposition);
-    }
-  }
+    };
+  }, [onReposition]);
 
-  componentWillUnmount() {
-    if (this.props.onReposition) {
-      Browser.getWindow().removeEventListener(
-        "resize",
-        this.props.onReposition
-      );
-      if (this.scrollParent) {
-        this.scrollParent.removeEventListener(
-          "scroll",
-          this.props.onReposition
-        );
-      }
-    }
-    this.defaultNode.remove();
-    this.defaultNode = null;
-    this.scrollParent = null;
-  }
+  useEffect(() => {
+    return () => {
+      portalNode.remove();
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
 
-  updateClassNames(portalClassName) {
-    this.defaultNode.removeAttribute("class");
-    this.defaultNode.setAttribute("class", portalClassName);
-    if (this.props.className) {
-      this.props.className.split(" ").forEach((el) => {
-        this.defaultNode.classList.add(el);
-      });
-    }
-  }
+  const addClassNames = (node) => {
+    className.split(" ").forEach((el) => {
+      node.classList.add(el);
+    });
 
-  getPortalDiv() {
+    return node;
+  };
+
+  const getPortalContainer = () => {
     const portalClassName = "carbon-portal";
-    if (
-      this.props.id !== undefined &&
-      Browser.getDocument().getElementById(this.props.id)
-    ) {
-      this.defaultNode = Browser.getDocument().getElementById(this.props.id);
+    let node = portalNode;
+
+    if (!node && id !== undefined && document.getElementById(id)) {
+      node = document.getElementById(id);
+      setPortalNode(node);
     }
 
-    if (
-      !this.defaultNode ||
-      (this.props.id !== undefined &&
-        !Browser.getDocument().getElementById(this.props.id))
-    ) {
-      this.defaultNode = Browser.getDocument().createElement("div");
-      this.defaultNode.classList.add(portalClassName);
-      this.defaultNode.setAttribute("data-portal-exit", this.guid);
-      if (this.props.id !== undefined) {
-        this.defaultNode.setAttribute("id", this.props.id);
+    if (!node || (id !== undefined && !document.getElementById(id))) {
```

Comment:
Could this be an `else if` or is there still a chance node could be undefined/ null?

Reply:
Yeah you're right it could be, I just copied this from the old Portal. Part of the second if condition was unnecessary as well so I've removed it now

Reply:
Yeah good shout